app/controllers/your-psp/yourpsp-tasks.service.js

```diff
@@ -0,0 +1,50 @@
+'use strict'
+
+function stripeSetupStageComplete (account, stage) {
+  if (account.connectorGatewayAccountStripeProgress) {
+    return account.connectorGatewayAccountStripeProgress[stage]
+  }
+  return false
+}
+
+function getNewStripeTaskList (targetCredential, account) {
+   if (targetCredential.payment_provider === 'stripe') {
+    return {
+      'ENTER_BANK_DETAILS': {
+        enabled: !stripeSetupStageComplete(account, 'bankAccount'),
+        complete: stripeSetupStageComplete(account, 'bankAccount')
+      },
+      'ENTER_RESPONSIBLE_PERSON': {
+        enabled: !stripeSetupStageComplete(account, 'responsiblePerson'),
+        complete: stripeSetupStageComplete(account, 'responsiblePerson')
+      },
+      'ENTER_DIRECTOR': {
+        enabled: !stripeSetupStageComplete(account, 'director'),
+        complete: stripeSetupStageComplete(account, 'director')
+      },
+      'ENTER_VAT_NUMBER': {
+        enabled: !stripeSetupStageComplete(account, 'vatNumber'),
+        complete: stripeSetupStageComplete(account, 'vatNumber')
+      },
+      'ENTER_COMPANY_NUMBER': {
+        enabled: !stripeSetupStageComplete(account, 'companyNumber'),
+        complete: stripeSetupStageComplete(account, 'companyNumber')
+      },
+      'CONFIRM_ORGANISATION_DETAILS': {
+        enabled: !stripeSetupStageComplete(account, 'organisationDetails'),
+        complete: stripeSetupStageComplete(account, 'organisationDetails')
+      },
+      'UPLOAD_GOVERNMENT_ENTITY_DOCUMENT': {
+        enabled:  !stripeSetupStageComplete(account, 'governmentEntityDocument'),
```

Comment:
Government entity task should only be enabled when all other tasks are completed. Can other tasks be checked?

Could be something like
```
enabled: stripeSetupStageComplete(account, 'bankAccount') &&
          stripeSetupStageComplete(account, 'responsiblePerson') &&
          ...  [rest of the tasks] ...   
          !stripeSetupStageComplete(account, 'governmentEntityDocument')
```

and additional tests

Reply:
Done too